[BE-199] 모아보기 최신레코드

kdomo opened the pull request:
## 관련 이슈 번호
- [BE-199 / 모아보기 최신레코드](https://recodeit.atlassian.net/browse/BE-199)

## 설명
모아보기 탭에서 최신레코드 반환하는 API입니다.
테스트 코드 작성 시 기존 예외만 테스트 하던 내용이 해당 메서드에서는 의미가 없다고 생각하여 mock처리 후 데이터 검증하였습니다.
Swagger `@ApiResponse`에서 response를 Page객체로 반환해주어야 했는데 `Page<responseDto>.class` 의 형태가 되지 않아 작성하지 않았더니 정상 응답 형태로 나옵니당.

## 변경사항
- [x] 최신 레코드 조회 테스트 코드 작성
- [x] RecentRecordRequest, Response Dto 생성
- [x] RecordService.java에서 getRecentRecord 메서드 생성 (최신 레코드 조회 비지니스 로직) 

## 질문사항
10개의 레코드를 가져온 후 각각 댓글의 카운트를 찾아야해서 `N+1` 문제 발생합니다. 좋은 방법이 있다면 알려주신다면 적극 반영하겠습니다.

kdomo committed:
[BE-199] test: 최신레코드 API 테스트 코드 작성

kdomo committed:
[BE-199] feat: 최신레코드 API

kdomo committed:
[BE-199] style: 코드 포맷팅 정상적으로 수정

Jaeyeop-Jung commented on src/main/java/com/recordit/server/repository/RecordRepository.java:
```diff
@@ -62,4 +62,7 @@ Page<Record> findAllByWriterAndCreatedAtBetweenOrderByCreatedAtDesc(
 			+ "order by RAND() limit :size", nativeQuery = true)
 	List<Record> findRandomRecordByRecordCategoryId(Integer size, Long categoryId);
+
+	@EntityGraph(attributePaths = {"recordIcon", "recordColor"})
+	Page<Record> findAll(Pageable pageable);
```
```suggestion
Page<Record> findAllFetchRecordIconAndRecordColor(Pageable pageable);
```
이렇게 가시죠

Jaeyeop-Jung commented on src/main/java/com/recordit/server/service/RecordService.java:
```diff
@@ -326,4 +328,21 @@ public MixRecordResponseDto getMixRecords() {
+				recentRecordRequestDto.getSize(),
+				Sort.Direction.DESC,
+				"createdAt"
+		));
```
```suggestion
		Page<Record> recordPage = recordRepository.findAll(
		        PageRequest.of(
				recentRecordRequestDto.getPage(),
				recentRecordRequestDto.getSize(),
				Sort.Direction.DESC,
				"createdAt"
		)
	);
```

들여쓰기 이렇게 가시죠

Jaeyeop-Jung commented on src/main/java/com/recordit/server/service/RecordService.java:
```diff
@@ -326,4 +328,21 @@ public MixRecordResponseDto getMixRecords() {
+		return recordPage.map(record -> RecentRecordResponseDto.of(
+				record,
+				commentRepository.countByRecordId(record.getId())
+		));
```
```suggestion
		return recordPage.map(
		        record -> RecentRecordResponseDto.of(
			        record,
			        commentRepository.countByRecordId(record.getId())
		        )
	        );
```

이렇게 가시죠

Jaeyeop-Jung commented on src/test/java/com/recordit/server/service/RecordServiceTest.java:
```diff
@@ -556,4 +563,70 @@ class RecordServiceTest {
+					.willReturn(new PageImpl<>(recordList, pageRequest, 1));
+			//when
+			Page<RecentRecordResponseDto> recentRecord = recordService.getRecentRecord(
+					recentRecordRequestDto);
```
```suggestion
					recentRecordRequestDto
				);
```

Jaeyeop-Jung reviewed (commented):
스타일 변경하는거랑 repository 쿼리 추가하는거만 하면 될거 같아요!

sookyeonghwang reviewed (commented):
우왕 도모짱

kdomo committed:
[BE-199] fix: 코드리뷰 후 수정
- Fetch join을 사용해서 findAll메서드를 사용하면 FetchType을 EAGER로 사용하는 것고 다름 없어, 메소드 명을 변경하였습니다.
- 포맷팅 수정 하였습니다.
